source/Outfit.h

```diff
@@ -120,6 +121,7 @@ class Outfit : public Weapon {
 	int index;
 	std::string description;
 	const Sprite *thumbnail = nullptr;
+	Body thumbnailBody;
```

Comment:
outfit should not contain a body, a body is an object that is floating/presented in flight

Reply:
This was the only way a Sprite could be loaded, as confirmed by Amazinite on #human-chat where I was talking to Arkenexus and ziproot about ideas I had had.

Perhaps some refactoring is needed to move sprite load out of Body, but that is an independent task.

Reply:
in that case that refactor should be done first

Reply:
I would also like to see image/animation properties and behaviour and location/orientation/movement properties and behaviour separated so only the former is necessary here (and some other places).

Reply:
@warp-core can you give an example of what your preferred syntax would look like?

Reply:
Looking at Body.h, I'm imagining something along the lines of:
- Most of the private fields (relating to sprites and animations) would be in their own class, along with the associated methods. Perhaps it could be called "Drawable", though this may be unclear because `Sprite`s themselves can be drawn without the extra properties.
- The other fields would probably remain in Body, which would extend the new class created from its other properties.
- Types that require the positional and image properties would see no change.
- Types that only require the image properties would only need to deal with the new class, not the whole Body class.

Reply:
Intuitivly, I think the frame rate, starting frame, etc. information should be part of a class called Sprite, which would align with the Node name used in the data files. Every Body currently has a Sprite, so just moving the fields into that class would work for the time being.

I prefer working in steps as small as possible, so I would rather move the fields first, then refactor as a follow-up PR.

After that, the question becomes, do we split Sprite back up again into a class with no animation information (specificly, just one frame). My intuition says the memory cost of the additional animation fields is minimal, and this may not be a big win. Sprites without an associated Body instance are (correct me if this is wrong) mostly used for panel backgrounds, borders and buttons. Sprites used for drawing the UI do not use SwizzleMask, for example, so that could go in the new class too.

A third choice is to rename Sprite to something (SpriteFrame?), and have the new Sprite class extend it.

Reply:
I do not recommend moving these fields into the existing "Sprite" class. Each "Sprite" object corresponds to one named image. That is, a collection of frames, high DPI frames, and normal and high DPI swizzle masks.
Each "Sprite" object is used by many "Body" objects, but each of those "Body"s is not required to use the "Sprite" with the same animation or scale properties. Additionally, even "Body"s that do have the same animation configurations may not start on the same "step". This is just not an approach that is going to work very easily, or at al.
I also don't recommend renaming "Sprite" and using that name for the new class; it's going to take an extremely good argument to convince me that any other name would be worse than changing every place that currently refers to the "Sprite" class.